components/core/src/clp_s/JsonFileIterator.cpp

```
@@ -118,6 +212,59 @@ bool JsonFileIterator::get_json(simdjson::ondemand::document_stream::iterator& i
+                        m_buf_occupied,
+                        simdjson::SIMDJSON_PADDING
+                );
+                m_buf_occupied = result.new_buf_occupied;
+                m_sanitization_bytes_added += m_buf_occupied - old_buf_occupied;
+
+                // Log sanitization details
+                if (!result.sanitized_char_counts.empty()) {
+                    size_t total_sanitized = 0;
+                    std::string char_details;
+                    for (auto const& [ch, count] : result.sanitized_char_counts) {
+                        if (!char_details.empty()) {
```

Cppcheck (2.19.0)

[style] The function 'sub_span' is never used.

(unusedFunction)